dynawo/sources/Modeler/Common/DYNDynamicData.cpp

```diff
@@ -274,6 +271,36 @@ DynamicData::getParametersSet(const string& modelId, const string& parFile, cons
   return parametersSetCollection->getParametersSet(parId);
 }
 
+void
+DynamicData::addReferencesFromParFilesInParSet(shared_ptr<BlackBoxModel> bbm, shared_ptr<ParametersSet> modelSet) const {
+  if (modelSet == nullptr)
+    return;
+  const std::vector<std::string> parReferences = modelSet->getParReferences();
+  for (const std::string parRefName : parReferences) {
+    const shared_ptr<Reference> parRef = modelSet->getReference(parRefName);
+    std::string parFile;
+    if (parRef->getParFile() == "") {
+      parFile = bbm->getParFile();
+    } else {
+      parFile = parRef->getParFile();
+    }
+    std::string parFilePath = canonical(parFile, rootDirectory_);
+    parameters::XmlImporter parametersImporter;
```

Comment:
This mean that we can read multiple times the same file which is a performance blackhole. Please use getParametersSet method